src/track/steminfoimporter.cpp

```diff
@@ -0,0 +1,139 @@
+    char buffer[kAtomHeaderSize];
+    while (!reader.atEnd() || (box_size && byteRead >= box_size)) {
+        reader.read(buffer, kAtomHeaderSize);
+        byteRead += box_size;
+        if (ATOM_TYPE(buffer + 4) == path[pathIdx]) {
+            return seekTillAtom(reader, path, ATOM_TYPE(buffer) - kAtomHeaderSize, pathIdx + 1);
+        }
+        reader.skip(ATOM_TYPE(buffer) - kAtomHeaderSize);
+    }
+    return -1;
+}
+
+} // anonymous namespace
+
+// static
+bool StemInfoImporter::isStemFile(
+        const QString& aFileName) {
+    return aFileName.endsWith(kStemFileExtension);
+}
+
+QList<StemInfo> StemInfoImporter::importStemInfos(
+        const QString& filePath) {
+    // Fetch the STEM manifest which contain stream details
+    auto file = QFile(filePath);
+    if (!file.open(QIODeviceBase::ReadOnly)) {
+        kLogger.warning()
+                << "Failed to open input file"
+                << filePath;
+        return {};
+    }
+
+    uint32_t manifestSize;
+    if (!(manifestSize = seekTillAtom(file, kStemManifestAtomPath))) {
+        kLogger.warning()
+                << "Failed to find the steam manifest in the file"
+                << filePath;
+        return {};
+    }
+
+    auto jsonData = QJsonDocument::fromJson(file.read(manifestSize));
+    VERIFY_OR_DEBUG_ASSERT(jsonData.isObject()) {
+        kLogger.warning()
+                << "Failed to extract the manifest data"
+                << filePath;
+        return {};
+    };
+
+    auto json = jsonData.object();
+
+    // Check the manifest version
+    auto version = json.value("version").toInteger(-1);
+    if (version <= 0) {
+        kLogger.debug() << "Unexpected or missing version value in STEM manifest";
+        return {};
+    } else if (version > kSupportedStemVersion) {
+        kLogger.debug() << "Unsupported stem version" << version << "but trying anyway";
+    }
+
+    // Extract stem metadata
+    auto stems = json.value("stems");
+    if (!stems.isArray()) {
+        kLogger.debug() << "Unexpected or missing stems value in STEM manifest";
+        return {};
+    }
+    auto stemArray = stems.toArray();
+    QList<StemInfo> stemsList;
+    stemsList.reserve(stemArray.size());
+    for (const auto& stemRef : stemArray) {
+        if (!stemRef.isObject()) {
+            kLogger.debug() << "Unexpected or missing stems value in STEM manifest";
+            return {};
+        }
+        auto stem = stemRef.toObject();
+        auto color = QColor(stem.value("color").toString());
+        auto name = stem.value("name").toString();
+        if (!color.isValid() || name.isEmpty()) {
+            kLogger.debug() << "Unexpected or missing stem name or attribute in STEM manifest";
+            return {};
+        }
+        stemsList.emplace_back(name, color);
+    }
+
+    // Extract DSP information
+    // TODO(XXX) DSP only contains limit and a compressor, which aren't
+    // supported by Mixxx yet. parse and implement when supported
```

Comment:
```suggestion
    // used by Mixxx yet. Parse and implement when supported
```

Reply:
I'm afraid this won't mean what I tried to say here. I have reworded it differently now, let me know if that makes more sense.

Reply:
Comment is clear now!